From: Thomas HAMEL <thomas-externe.hamel@edf.fr>
Date: Tue, 10 Apr 2018 18:00:08 +0200
Subject: slurmctld: Check UID in pack_job before hiding

Checking hiding is very complex, so this patch do the very simple
UID job check before the complex check.
---
 src/slurmctld/job_mgr.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/slurmctld/job_mgr.c b/src/slurmctld/job_mgr.c
index c225916..3b3a891 100644
--- a/src/slurmctld/job_mgr.c
+++ b/src/slurmctld/job_mgr.c
@@ -9279,6 +9279,10 @@ static void _pack_job(struct job_record *job_ptr,
 {
 	xassert (job_ptr->magic == JOB_MAGIC);
 
+	if ((pack_info->filter_uid != NO_VAL) &&
+	    (pack_info->filter_uid != job_ptr->user_id))
+		return;
+
 	if (((pack_info->show_flags & SHOW_ALL) == 0) &&
 	    (pack_info->uid != 0) &&
 	    _all_parts_hidden(job_ptr, pack_info->uid))
@@ -9287,10 +9291,6 @@ static void _pack_job(struct job_record *job_ptr,
 	if (_hide_job(job_ptr, pack_info->uid, pack_info->show_flags))
 		return;
 
-	if ((pack_info->filter_uid != NO_VAL) &&
-	    (pack_info->filter_uid != job_ptr->user_id))
-		return;
-
 	pack_job(job_ptr, pack_info->show_flags, pack_info->buffer,
 		 pack_info->protocol_version, pack_info->uid);
 
